cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceLifecycleContext.java

```diff
@@ -126,6 +127,16 @@ public String getNamespace() {
   }
 
   @Override
+  public void addLocalFile(URI uri) {
+    throw new UnsupportedOperationException("Local cache files are not supported in Mapper/Reducer tasks.");
```

Comment:
But the contract of this class is to delegate. The context that this delegate to will do the right thing.